### DISCUSSION TITLE
embeddings RoadMap ?

### DISCUSSION THREAD

**Discussion by iwaitu** [microsoft/onnxruntime-genai#713] [Ideas]
Is there a plan on the RoadMap to support text embeddings? Similar to Ollama, where in Python I can do:
```
import ollama
model_id = "llama"
result = ollama.embeddings(model=model_id, prompt="test embedding")
```
It would be great if GenAI could add an embeddings object.

**Comment by kunal-vaishnavi** [comment]
The embeddings (a.k.a. last hidden states) are [now accessible](https://github.com/microsoft/onnxruntime-genai/pull/1127) via ONNX Runtime GenAI. You can create the ONNX model to output the hidden states using [ONNX Runtime GenAI's model builder](https://github.com/microsoft/onnxruntime-genai/blob/main/src/python/py/models/README.md#include-last-hidden-states-output) with `--extra_options include_hidden_states=true`.
